pkg/cloud/azure/actuators/machine/actuator.go

```diff
@@ -105,6 +106,7 @@ func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machi
 			klog.Errorf("Error storing machine info: %v", err)
 		}
 		a.handleMachineError(machine, apierrors.CreateMachine("failed to reconcile machine %qs: %v", machine.Name, err), createEventAction)
```

Comment:
what if we pass invalid input to the API request which make `err = a.reconcilerBuilder(scope).Create(context.Background())` fail?

Reply:
@enxebre it shouldn't get to this point or am I wrong here. The `NewMachineScope` should catch invalid configuration and if an error gets thrown from `Create()` it should be retry-able?

Reply:
NewMachineScope validate in terms of decoding types. It does not do semantic validation e.g user input an instance type which does not exists. We should fail with invalidConfig when the API request returns a 4* http code